include/qthread/qthread.h

```diff
@@ -87,7 +87,7 @@ using std::memory_order_relaxed;
 
 #include "macros.h"
 
-#define MACHINE_FENCE atomic_thread_fence(memory_order_acq_rel);
```

Comment:
Is change is needed for correctness or for performance?

Reply:
Surprisingly, performance. I'm still confused as to how/why this is happening, but for the chapel benchmark I'm using to track the regression there's a modest performance improvement from using the sequentially consistent fences instead of just acquire release consistent. This is consistent with what the old assembly based version was doing prior to https://github.com/sandialabs/qthreads/commit/b122234e164be9bc6bc1cd3d45673e53c7f22f95 anyway, but it's still weird.

Reply:
I'm not aware of anything in our codebase that ought to actually require sequential consistency for correctness' sake. The fact that sequential consistency is beneficial here seems to indicate that there's something suboptimal going on internally. This patch at least corrects the downstream performance regression though.